File: 0844-backspace-string-compare/0844-backspace-string-compare.py
class Solution(object):
    def backspaceCompare(self, s, t):
        s1=""
        for i in s:
            if i.isalpha():
                s1+=i
            else:
                if s1:
                    s1=s1[:-1]
        t1=""
        for i in t:
            if i.isalpha():
                t1+=i
            else:
                if t1:
                    t1=t1[:-1]
        return s1==t1

File: 0844-backspace-string-compare/0844-backspace-string-compare.java
class Solution {
    public static boolean backspace(String s, String t) {
        Stack<Character> stack = new Stack<>();
        for (char c : s.toCharArray()) {
            if (c == '#' && !stack.isEmpty()) {
                stack.pop();
            } else if (Character.isLetter(c)) {
                stack.push(c);
            }
        }

        Stack<Character> stack1 = new Stack<>();
        for (char c : t.toCharArray()) {
            if (c == '#' && !stack1.isEmpty()) {
                stack1.pop();
            } else if (Character.isLetter(c)) {
                stack1.push(c);
            }
        }

        StringBuilder sbStack = new StringBuilder();
        while (!stack.isEmpty()) {
            sbStack.insert(0, stack.pop());
        }

        StringBuilder sbStack1 = new StringBuilder();
        while (!stack1.isEmpty()) {
            sbStack1.insert(0, stack1.pop());
        }

        return sbStack.toString().equals(sbStack1.toString());
    }
    public boolean backspaceCompare(String s, String t) {
        return backspace(s, t);
    }
}